src/databricks/sqlalchemy/dialect.py

```diff
@@ -1,52 +1,301 @@
+    # Once information constraint is complete, this will need to be implemented.
+    # This is needed for Connection.create_all()
+    def create_table_constraints(
+        self, table, _include_foreign_key_constraints=None, **kw
+    ):
+        return []
+
+
+# I started with the following lookup table (indexed by DATA_TYPE) and it is rather nice since Decimal can be detected directly.
+# However, as DATA_TYPE values are rather obtuse, I switched to use COLUMN_TYPE_NAME instead (the table below)
+# _type_map = {
+#     -6: types.Integer,        # tiny_int
+#     5: types.Integer,         # small_int
+#     4: types.Integer,         # int
+#     -5: types.BigInteger,     # big_int
+#     6: types.Float,
+#     3: types.DECIMAL,
+#     16: types.Boolean,
+#     12: types.String,
+#     91: DatabricksDate,       # date
+#     93: DatabricksTimestamp,  # timestamp
+#     1111: interval
+# }
+
+
+# This lookup is by TYPE_NAME which is easier to maintain and likely safer in the long term.
+# NB: Decimal is explicitly excluded here as each occurence's TYPE_NAME includes the occurence's precision and scale
+#     See/refer to COLUMN_TYPE_DECIMAL below.
+
+# this map SQL types onto Python representation; note the deliberate omission of Decimal!
+_type_map = {
+    "TINYINT": types.Integer,  # tiny_int
+    "SMALLINT": types.Integer,  # small_int
+    "INT": types.Integer,  # int
+    "BIGINT": types.BigInteger,  # big_int
+    "FLOAT": types.Float,
+    "DOUBLE": types.Float,  # double fits into a Python float
+    "BOOLEAN": types.Boolean,
+    "STRING": types.String,
+    "DATE": types.DATE,  # date
+    "TIMESTAMP": types.TIMESTAMP,  # timestamp
+}
+# this is used to match a column's DATA_TYPE for Decimal; it will map to types.DECIMAL
+COLUMN_TYPE_DECIMAL = 3
+# COLUMN_TYPE_INTERVAL=1111
+
+
+class DatabricksDialect(default.DefaultDialect):
     # Possible attributes are defined here: https://docs.sqlalchemy.org/en/14/core/internals.html#sqlalchemy.engine.Dialect
     name: str = "databricks"
     driver: str = "thrift"
     default_schema_name: str = "default"
 
+    preparer = DatabricksIdentifierPreparer
+
+    # TODO: revisit server-side cursors
+    # ref: https://docs.databricks.com/dev-tools/python-sql-connector.html#manage-cursors-and-connections
+    execution_ctx_cls = default.DefaultExecutionContext
+
+    statement_compiler = compiler.SQLCompiler
+    ddl_compiler = DatabricksDDLCompiler
+    type_compiler = DatabricksTypeCompiler
+
+    # the following attributes are cribbed from HiveDialect:
+    supports_views = False
+    supports_alter = True
+    supports_pk_autoincrement = False
+    supports_default_values = False
+    supports_empty_insert = False
+    supports_native_decimal = True
+    supports_native_boolean = True
+    supports_unicode_statements = True
+    supports_unicode_binds = True
+    returns_unicode_strings = True
+    description_encoding = None
```

Comment:
what's description_encoding?

Reply:
According to what I found in SA's doc (https://github.com/sqlalchemy/sqlalchemy/blob/main/doc/build/changelog/changelog_05.rst), it "is used for encoding the column name when processing the metadata. This usually defaults to utf-8." I went with "None" as for all the other dialects I studied.